File: common/src/main/java/net/particify/arsnova/common/uuid/UuidToStringJsonConverter.java
package net.particify.arsnova.common.uuid;

import java.util.UUID;
import tools.jackson.databind.JavaType;
import tools.jackson.databind.type.TypeFactory;
import tools.jackson.databind.util.StdConverter;

public class UuidToStringJsonConverter extends StdConverter<UUID, String> {

  @Override
  public String convert(final UUID value) {
    return UuidHelper.uuidToString(value);
  }

  @Override
  public JavaType getInputType(final TypeFactory typeFactory) {
    return typeFactory.constructType(UUID.class);
  }

  @Override
  public JavaType getOutputType(final TypeFactory typeFactory) {
    return typeFactory.constructType(String.class);
  }
}

File: authz/src/main/kotlin/net/particify/arsnova/authz/model/serialization/UuidToStringJsonConverter.kt
package net.particify.arsnova.authz.model.serialization

import com.fasterxml.jackson.databind.JavaType
import com.fasterxml.jackson.databind.type.TypeFactory
import com.fasterxml.jackson.databind.util.Converter
import net.particify.arsnova.authz.model.serialization.UuidHelper.uuidToString
import java.util.UUID

class UuidToStringJsonConverter : Converter<UUID, String?> {
  override fun convert(value: UUID): String? {
    return uuidToString(value)
  }

  override fun getInputType(typeFactory: TypeFactory): JavaType {
    return typeFactory.constructType(UUID::class.java)
  }

  override fun getOutputType(typeFactory: TypeFactory): JavaType {
    return typeFactory.constructType(String::class.java)
  }
}
